p2p/src/sync/peer.rs

```diff
@@ -254,22 +386,36 @@ where
 
     async fn request_headers(&mut self) -> Result<()> {
         let locator = self.chainstate_handle.call(|this| this.get_locator()).await??;
-        debug_assert!(locator.len() <= *self.p2p_config.msg_max_locator_count);
+        if locator.len() > *self.p2p_config.msg_max_locator_count {
+            // Note: msg_max_locator_count is not supposed to be configurable outside of tests,
+            // so we should never get here in production code. Moreover, currently it's not
+            // modified even in tests. TODO: make it a constant.
+            log::warn!(
+                "[peer id = {}] Sending locator of the length {}, which exceeds the maximum length {:?}",
+                self.id(),
+                locator.len(),
+                self.p2p_config.msg_max_locator_count
+            );
+        }
 
-        log::trace!("Sending header list request to {} peer", self.id());
+        log::debug!("[peer id = {}] Sending header list request", self.id());
```

Comment:
Btw, I wanna mention a very important point about peer ids, because there's some legacy that has been ruining logging for that. A peer id doesn't really mean anything. Logging with peer id = X is completely meaningless, because that number comes from a counter. The correct behavior is to put identifying information about the peer in the logs, like, for example, their ip address.